07.ls_object/file_detail.rb

```diff
@@ -34,24 +34,31 @@ def initialize(name, directory = '')
              "#{directory}/#{name}"
            end
 
-    stat = File::Stat.new(path)
-    @type = TYPE_LIST[stat.ftype.to_sym]
-    @mode = fetch_file_mode(stat)
-    @nlink = stat.nlink
-    @uid = Etc.getpwuid(stat.uid).name
-    @gid = Etc.getgrgid(stat.gid).name
-    @size = stat.size
-    @mtime = stat.mtime
-    @blocks = stat.blocks
+    @stat = File::Stat.new(path)
+  end
+
+  def type
+    TYPE_LIST[@stat.ftype.to_sym]
+  end
+
+  def mode
+    get_file_mode(@stat)
+  end
+
+  def uid
+    Etc.getpwuid(@stat.uid).name
+  end
+
+  def gid
+    Etc.getgrgid(@stat.gid).name
   end
 
   private
 
-  def fetch_file_mode(stat)
-    result = []
+  def get_file_mode(stat)
     # ファイルモードを8進数に変換して末尾3桁（パーミッション）を取得する
-    stat.mode.to_s(8)[-3..].each_char do |char|
-      result << MODE_LIST[char.to_i]
+    result = stat.mode.to_s(8)[-3..].each_char.map do |char|
+      MODE_LIST[char.to_i]
     end
     result.join
```

Comment:
```suggestion
    stat.mode.to_s(8)[-3..].each_char.map do |char|
      MODE_LIST[char.to_i]
    end.join
```

こういう書き方もありといえばありです。（好みは分かれるかも）